### PR TITLE
Add sidebar for level builder

### PR DESCRIPTION
**Pull request by kobryan0619**
Allows for levelbuilders to more easily navigate the level edit page by adding a sidebar for navigation.

Context:
- Navigating level builder to edit levels is a bit clunky.  
- [Dan had created a google Chrome extension](https://codedotorg.slack.com/archives/C026S5P8QRM/p1684192958150849) that created a sidebar that folks on the T&L team had been using to help navigate.  This PR does much of the same thing as the Chrome extension.
- [Ken and I had done an initial stab](https://github.com/code-dot-org/code-dot-org/pull/56958) at this work by adding Javascript into a HAML page.  After feedback, Ken and I tried to move this to React in this PR. 

Note that currently, level builder is not responsive to resizing the page so that the editor says within the screen view. 

The NEW video below shows the navigation and how the page reacts to re-sizing.

https://github.com/code-dot-org/code-dot-org/assets/24883357/e56ed186-21c2-4024-909f-d25f985a62aa

## Testing story

[Ran mini-bug-bash](https://docs.google.com/document/d/1jdWG0XSPfrW5J1Hu7O757rqisMn-hfKCRPdMmZoptek/edit#heading=h.fmdijfi7c3pz) with members of the T&L team + Hannah B + Turner.  From that came a lot of requests to expand the scope of this project, and while we skipped the big scope-changing requests, we did implement a request to have the sidebar include indented links to sub-editors on the page.

### CODE REVIEW

**Review by kakiha11 — COMMENTED** [review]
Thanks for all this work to get this working with the React component Kaitie!

File: apps/src/sites/studio/pages/levels/editors/_navigation_bar.js
Comment (breville) [inline apps/src/sites/studio/pages/levels/editors/_navigation_bar.js]: Can remove this before merging?

File: dashboard/app/views/levels/edit.html.haml
Comment (breville) [inline dashboard/app/views/levels/edit.html.haml]: Much easier to review with the `Hide whitespace` option checked (i.e. `&w=1` in the URL)! :)

**Review by breville — APPROVED** [review]
This is great!  Thanks for incorporating the feedback on the previous PR.

It looks like there are just a few linter issues that drone picked up that should be resolved before merging.

It's also interesting that the component is called `NavigationBar` but the PR title refers to a "sidebar".  Sometimes it's not until writing up the PR that we notice the best name for a component, so it's possible that a component name incorporating "sidebar" could be a useful name going forward, but that's just some food for thought.

**Comment by dju90** [comment]
This looks fantastic. A couple dotting-i's-crossing-t's questions:
- Is there an easy way to roll this back in the unlikely event that it breaks the level editing experience?
- Are there any other risks to shipping this that we should be aware of?
- Does this change work across the major browsers? Chrome, Safari, Edge

**Comment by davidsbailey** [comment]
thank you for taking the time to extract this into js/jsx! I don't think we have great automated testing on level edit pages, and this change affects many level edit pages. so my feedback would be to make sure you test that nothing is majorly broken on level edit pages for a good variety of level types.

**Review by davidsbailey — COMMENTED** [review]
LGTM once other level types are tested

File: apps/src/lab2/levelEditors/NavigationSidebar.jsx
Comment (davidsbailey) [inline apps/src/lab2/levelEditors/NavigationSidebar.jsx:1]: Sorry for not suggesting this earlier, but I think it would be worthwhile to add one unit test which renders a sidebar. it looks like you need there to be some elements already in the DOM, which you can create like this: https://github.com/code-dot-org/code-dot-org/blob/f836853e7e87a18631ac25a2d93254ddb7f5d9d1/apps/test/unit/code-studio/levels/freeResponse.js#L18-L21

I think it would be fine to just make sure the component can render with a single non-nested link -- the goal is to make sure we have some automated test coverage to prevent a refactor of this component (think jest migration, etc) from completely breaking the level edit page.

**Review by davidsbailey — APPROVED** [review]
I tried this out and it is really nice! LGTM after adding a simple unit test for the react component.

File: apps/src/lab2/levelEditors/NavigationSidebar.jsx
Comment (kakiha11) [reply apps/src/lab2/levelEditors/NavigationSidebar.jsx:1]: Hey Dave, I've added this test in the latest commit. Let me know what you think!

**Comment by davidsbailey** [comment]
nice work Ken! LGTM 👍

**Pull request merged**
